scheduler/src/cook/mesos/task.clj

```diff
@@ -0,0 +1,167 @@
+(ns cook.mesos.task
+  (:require [clojure.algo.generic.functor :refer (fmap)]
+            [clojure.tools.logging :as log])
+  (import com.netflix.fenzo.TaskAssignmentResult))
+
+
+(defn TaskAssignmentResult->task-info
+  "Organizes the info Fenzo has already told us about the task we need to run"
+  [^TaskAssignmentResult fenzo-result]
+  (merge (:task-info (.getRequest fenzo-result))
+         {:ports-assigned (.getAssignedPorts fenzo-result)}))
+
+
+(defmulti combine-like-resources
+  (fn [list] (-> list first :type)))
+
+(defmethod combine-like-resources :value-scalar [resources]
+  (->> resources (map :scalar) (apply +)))
+
+(defmethod combine-like-resources :value-ranges [resources]
+  (->> resources (map :ranges) (apply concat) vec))
+
+(defn resources-by-role
+  "Given a set of offers, combine all the available resources into a comprehensible
+  and easily walkable data strucuture, grouped first by resource name e.g. mem, then
+  by role."
+  [offers]
+  (->> offers
+       (map :resources)
+       flatten
+       (group-by :name)
+       (fmap (fn [resources]
```

Comment:
Use `map-vals` from [prismatic-plumbing](https://github.com/plumatic/plumbing/blob/master/src/plumbing/core.cljx#L68) here. I prefer to avoid fmap unless the data you are getting really can take different forms and you are unsure what you are getting. Since in this case and the one just below we know it is a map and we want to map on the values, using `map-vals` is more clear what is occurring.